mentat/conversation.py

```diff
@@ -52,7 +59,40 @@ def get_model_response(self, config: ConfigManager) -> (str, list[CodeChange]):
         messages = self.messages.copy()
 
         code_message = self.code_file_manager.get_code_message()
-        messages.append({"role": "system", "content": code_message})
+        system_message = code_message
+
+        if self.code_map:
+            system_message_token_count = count_tokens(system_message)
+            messages_token_count = 0
+            for message in messages:
+                messages_token_count += count_tokens(message["content"])
+            token_buffer = 1000
+            token_count = (
+                system_message_token_count + messages_token_count + token_buffer
+            )
+            max_tokens_for_code_map = self.token_limit - token_count
+
+            if self.code_map.token_limit:
+                code_map_message_token_limit = min(
+                    self.code_map.token_limit, max_tokens_for_code_map
+                )
+            else:
+                code_map_message_token_limit = max_tokens_for_code_map
+
+            code_map_message = self.code_map.get_message(
+                token_limit=code_map_message_token_limit
+            )
+            if code_map_message:
+                system_message += f"\n{code_map_message}"
+            else:
+                cprint_message = [
+                    "\nExcluding CodeMap from system message.",
+                    "Reason: not enough tokens available in model context.",
+                ]
+                cprint_message = "\n".join(cprint_message)
+                cprint(cprint_message, color="yellow")
```

Comment:
Let's have CodeMap.get_message also return which level map it sent back (full detail with signatures, without signatures, file map only), and inform the user which level is being used.

I'm not sure it'll be something we'll always show the user, but I think it's important for now, because users might notice different quality edits and not know it's due to the map changing (and that removing extra files from context could have helped by giving more space for the map)